src/component/color.ts

```diff
@@ -0,0 +1,248 @@
+
+/* The color can be specified as
+
+  * a hexadecimal RGB value: #faf or #ffaaff
+  * a RGB value: rgb(255, 160, 255) or rgb(100%, 62.5%, 100%)
+   Each value is from 0 to 255, or from 0% to 100%.
+  * a RGBA value: rgba(255, 160, 255, 1) or rgba(100%, 62.5%, 100%, 1)
+    This variant includes an “alpha” component to allow specification of the opacity of a color.
+    Values are in the range 0.0 (fully transparent) to 1.0 (fully opaque).
+  * a HSL value: hsl(0, 100%, 50%)
+    A triple (hue, saturation, lightness).
+    hue is an angle in degrees. saturation and lightness are percentages (0-100%).
+  * a HSLA value: hsla(0, 100%, 50%, 1)
+    This variant includes an “alpha” component to allow specification of the opacity of a color.
+    Values are in the range 0.0 (fully transparent) to 1.0 (fully opaque).
+*/
+
+export type Color<T extends string> = ColorValue<T> | HTMLColor | 'currentColor' | 'transparent' | 'inherit';
```

Comment:
**chore:** 
As we need to add genericity on all classes, we choose to not implement it for now.
But we can add 'inherit', 'swimlane' in a first step, if we have time to test them.